WordPress/Sniffs/VIP/SuppressFiltersSniff.php

```diff
@@ -0,0 +1,192 @@
+		// eg.  $args = array( 'foo' => 'bar' ), $args['foo'] => 'bar'.
+		if ( T_VARIABLE === $this->tokens[ $argument_data ]['code'] ) {
+
+			$start = 0;
+
+			// Check if we are in a function.
+			$function = $this->phpcsFile->getCondition( $stackPtr, T_FUNCTION );
+
+			// If so, we check only within the function, otherwise the whole file.
+			if ( false !== $function ) {
+				$start = $this->tokens[ $function ]['scope_opener'];
+			} else {
+				// Check if we are in a closure.
+				$closure = $this->phpcsFile->getCondition( $stackPtr, T_CLOSURE );
+
+				// If so, we check only within the closure.
+				if ( false !== $closure ) {
+					$start = $this->tokens[ $closure ]['scope_opener'];
+				}
+			}
+
+			$end = $stackPtr;
+
+			// Walking through the tokens.
+			for ( $i = ( $start + 1 ); $i < $end; $i ++ ) {
+
+				if ( $this->tokens[ $i ]['content'] === $this->tokens[ $argument_data ]['content'] ) {
+
+					// Move to the next pointer where '=>' or ']' is placed.
+					$varData = $this->phpcsFile->findNext( array( T_CLOSE_SQUARE_BRACKET, T_DOUBLE_ARROW, T_ARRAY, T_OPEN_SHORT_ARRAY ),  $i );
+
+					if ( in_array( $this->tokens[ $varData ]['code'], array( T_CLOSE_SQUARE_BRACKET, T_DOUBLE_ARROW ),true ) ) {
+
+						$operator = $varData; // T_DOUBLE_ARROW.
+
+						if ( T_CLOSE_SQUARE_BRACKET === $this->tokens[ $varData ]['code'] ) {
+							$operator = $this->phpcsFile->findNext( T_EQUAL, ( $varData ) );
+						}
+
+						$keyIdx = $this->phpcsFile->findPrevious( array( T_WHITESPACE, T_CLOSE_SQUARE_BRACKET ), ( $operator - 1 ),null,true );
+
+						if ( ! is_numeric( $this->tokens[ $keyIdx  ]['content'] ) ) {
+							$key            = $this->strip_quotes( $this->tokens[ $keyIdx ]['content'] );
+							$valStart       = $this->phpcsFile->findNext( array( T_WHITESPACE ), ( $operator + 1 ), null, true );
+							$valEnd         = $this->phpcsFile->findNext( array( T_COMMA, T_SEMICOLON ), ( $valStart + 1 ), null, false, null, true );
+							$val            = $this->phpcsFile->getTokensAsString( $valStart, ( $valEnd - $valStart ) );
+							$val            = $this->strip_quotes( $val );
+							$array_value = $val;
+						}
+					} elseif ( in_array( $this->tokens[ $varData ]['code'], array( T_ARRAY, T_OPEN_SHORT_ARRAY ), true ) ) {
+						// Store array data into variable so that we can proceed later.
+						$param_items = $this->get_function_call_parameters( $varData );
+					}
+				}
+			}
+		} elseif ( T_ARRAY === $this->tokens[ $argument_data ]['code'] || T_OPEN_SHORT_ARRAY === $this->tokens[ $argument_data ]['code'] ) {
+
+			// It covers multiple array variable directly passed to function.
+			// eg. get_posts( array( 'foo' => 'bar', 'baz' => 'quux' ) ).
+			$param_items = $this->get_function_call_parameters( $argument_data );
+
+		} elseif ( T_CONSTANT_ENCAPSED_STRING === $this->tokens[ $argument_data ]['code'] || T_DOUBLE_QUOTED_STRING === $this->tokens[ $argument_data ]['code'] ) {
```

Comment:
You can simplify this by using `isset( Tokens::$stringTokens[ $this->tokens[ $argument_data ]['code'] ] )`